File: sme-creditmind-ai-backend/app/schemas/merchant.py
from __future__ import annotations

from typing import Optional, Literal, List

from pydantic import BaseModel


class CreditScoreSummary(BaseModel):
    score: int
    grade: Literal["A", "B", "C", "D", "E"]
    riskLevel: Literal["VERY_LOW", "LOW", "MEDIUM", "HIGH", "VERY_HIGH"]
    preApprovedLimit: float
    recommendation: Literal["APPROVE", "REVIEW", "DECLINE"]


class MerchantOut(BaseModel):
    id: str
    name: str
    businessRegNo: Optional[str] = None
    taxId: Optional[str] = None
    category: str
    subCategory: Optional[str] = None
    address: Optional[str] = None
    district: Optional[str] = None
    city: str
    posProvider: str
    contactName: Optional[str] = None
    contactPhone: Optional[str] = None
    contactEmail: Optional[str] = None
    status: Literal["ACTIVE", "INACTIVE", "SUSPENDED"]
    monthsActive: int
    createdAt: str
    updatedAt: str
    latestScore: Optional[CreditScoreSummary] = None


class MerchantListResponse(BaseModel):
    merchants: List[MerchantOut]
    total: int


class MerchantCreate(BaseModel):
    name: str
    businessRegNo: Optional[str] = None
    taxId: Optional[str] = None
    category: str
    subCategory: Optional[str] = None
    address: Optional[str] = None
    district: Optional[str] = None
    city: str = "Ho Chi Minh City"
    posProvider: str
    contactName: Optional[str] = None
    contactPhone: Optional[str] = None
    contactEmail: Optional[str] = None

File: sme-creditmind-ai/src/types/merchant.ts
export type MerchantStatus = "ACTIVE" | "INACTIVE" | "SUSPENDED";

export interface Merchant {
  id: string;
  name: string;
  businessRegNo?: string;
  taxId?: string;
  category: string;
  subCategory?: string;
  address?: string;
  district?: string;
  city: string;
  posProvider: string;
  contactName?: string;
  contactPhone?: string;
  contactEmail?: string;
  status: MerchantStatus;
  monthsActive: number;
  createdAt: string;
  updatedAt: string;
  latestScore?: CreditScoreSummary;
}

export interface CreditScoreSummary {
  score: number;
  grade: CreditGrade;
  riskLevel: RiskLevel;
  preApprovedLimit: number;
  recommendation: Recommendation;
}

export type CreditGrade = "A" | "B" | "C" | "D" | "E";
export type RiskLevel = "VERY_LOW" | "LOW" | "MEDIUM" | "HIGH" | "VERY_HIGH";
export type Recommendation = "APPROVE" | "REVIEW" | "DECLINE";
